pkg/session_recording/recordings.jsx

```diff
@@ -505,15 +484,17 @@
         }
 
         handleDateSinceChange(date) {
-            this.setState({dateSince: date});
-            this.clearRecordings();
-            this.journalctlRestart();
+            this.setState({dateSince: date}, () => {
+                this.clearRecordings();
+                this.journalctlRestart();
+            });
         }
 
         handleDateUntilChange(date) {
-            this.setState({dateUntil: date});
-            this.clearRecordings();
-            this.journalctlRestart();
+            this.setState({dateUntil: date}, () => {
+                this.clearRecordings();
+                this.journalctlRestart();
+            });
```

Comment:
From my current understanding of React - I see that using this callback is proper way, cause I've had issues with text input and async work of setState ( like previous state was read by app ), so because of datepicker and jQuery it might a bit change how it's working, but still it's how it should be - it's some kind similar to other JS frameworks ( like $http in AngularJS and using .then ).

I will check and read how components are getting update exactly. It might be that componentDidUpdate is getting called after setState changed ( guessing here ).